src/game/actionOptions/builder.js

```diff
@@ -38,6 +41,19 @@ const placeBuildingPlayer = ({ building, row, col }) => player => ({
   ],
 })
 
+// TODO specialized to farmers Inn => code must be moved!
+const applyImmediateAction = locs => player => {
+  const land = JSON.parse(JSON.stringify(player.land))
+  locs.forEach(loc => {
+    land[loc[0]][loc[1]].type = 'wood'
+    land[loc[0]][loc[1]].contents = []
+  })
```

Comment:
* instead of `loc => {`, can you see if you can unpack the parameter with a destructuring assignment like `([row,col]) =>`?
* the type should be `'forest'`, and not `'wood'`.
* while there shouldn't be anything in contents for the field anyway, nor any other fields right now, i feel like all this should do is change `type`, and leave all of the other keys as they are.

could be

```javascript
locs.forEach(([row, col]) => {
  land[row][col] = {
    ...land[row][col],
    type: 'forest'
  }
}

Reply:
Ok, I'll keep that in mind in the future.